src/modules/ekf2/EKF/position_fusion.cpp

```diff
@@ -181,3 +164,11 @@ void Ekf::resetHorizontalPositionToLastKnown()
 	// Used when falling back to non-aiding mode of operation
 	resetHorizontalPositionTo(_last_known_pos.xy(), sq(_params.pos_noaid_noise));
 }
+
+void Ekf::resetHorizontalPositionToExternal(const Vector2f &new_horiz_pos, float horiz_accuracy)
```

Comment:
I would just do this in place so the INFO message can contain more specific context.

Note you already have INFO both here and before the call.
![image](https://github.com/PX4/PX4-Autopilot/assets/84712/b8db396b-4b3a-4fa2-aa8e-570e9bc016ee)

Your first case is fine (""reset position to external observation"), but the second case you could have more context like "external position correction rejected, resetting". It may or may not be useful, but at a minimum it helps trace the execution path.